vm/src/vm/thread.rs

```diff
@@ -93,9 +93,9 @@ impl ThreadedVirtualMachine {
     /// to the parent thread and then `join()` on the `JoinHandle` (or similar), there is a possibility that
     /// the current thread will panic as `PyObjectRef`'s `Drop` implementation tries to run the `__del__`
     /// destructor of a python object but finds that it's not in the context of any vm.
-    pub fn run<F, R>(self, f: F) -> R
+    pub fn run<F, R>(&self, mut f: F) -> R
     where
-        F: FnOnce(&VirtualMachine) -> R,
```

Comment:
Is keeping FnOnce possible?

Reply:
It appears to compile fine with `FnOnce`, though I'm also wondering if perhaps someone in the future maybe stores an `Fn` or `FnMut` in their type, which would make using this function impossible in such a case. It kind of feels like it goes against the "shared" aspect of it.

Reply:
I added it back as `FnOnce` now